components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/model/Authentication.java

```diff
@@ -233,4 +236,62 @@ public Authentication build() {
             return new Authentication(this);
         }
     }
+
+    /**
+     * This builder build endpoint by taking the authentication type and properties as input.
+     */
+    public static class AuthenticationBuilder {
+
+        private Type authType;
+        private Map<String, String> authPropertiesMap;
+
+        public AuthenticationBuilder type(Type type) {
+
+            this.authType = type;
+            return this;
+        }
+
+        public AuthenticationBuilder properties(Map<String, String> authPropertiesMap) {
+
+            this.authPropertiesMap = authPropertiesMap;
+            return this;
+        }
+
+        public Authentication build() {
+
+            switch (authType) {
+                case BASIC:
+                    return new Authentication.BasicAuthBuilder(
+                            getProperty(Type.BASIC, authPropertiesMap, Property.USERNAME.getName()),
+                            getProperty(Type.BASIC, authPropertiesMap, Property.PASSWORD.getName())).build();
+                case BEARER:
+                    return new Authentication.BearerAuthBuilder(
+                            getProperty(Type.BEARER, authPropertiesMap, Property.ACCESS_TOKEN.getName())).build();
+                case API_KEY:
+                    return new Authentication.APIKeyAuthBuilder(
+                            getProperty(Type.API_KEY, authPropertiesMap, Property.HEADER.getName()),
+                            getProperty(Type.API_KEY, authPropertiesMap, Property.VALUE.getName())).build();
+                case NONE:
+                    return new Authentication.NoneAuthBuilder().build();
+                default:
+                    throw new IllegalArgumentException(String.format("An invalid authentication type '%s' is " +
+                            "provided for the authentication configuration of the endpoint.", authType.name()));
+            }
+        }
+
+        private String getProperty(Authentication.Type authType,  Map<String, String> actionEndpointProperties,
+                                   String propertyName) {
+
+            if (actionEndpointProperties != null && actionEndpointProperties.containsKey(propertyName)) {
+                String propValue = actionEndpointProperties.get(propertyName);
+                if (StringUtils.isNotBlank(propValue)) {
+                    return propValue;
+                }
+                throw new IllegalArgumentException(String.format("The Property %s cannot be blank.", propertyName));
+            }
+
+            throw new NoSuchElementException(String.format("The property %s must be provided as an authentication " +
+                    "property for the %s authentication type.", propertyName, authType.name()));
+        }
```

Comment:
is this right ? 
Shouldn't the message be something like "Property %s is not applicable for authentication type %s". Thus it should be illegal argument exception rather than NoSuchElementException?

Reply:
This error throws if the property map is null or the required property (eg: username for basic) is missing.

Reply:
make sense.